src/main/java/javax/validation/Path.java

```diff
@@ -198,6 +200,20 @@
 	 * @since 1.1
 	 */
 	interface BeanNode extends Node {
+
+		/**
+		 * @return the class of the container for the element affected by the violated constraint
+		 *
+		 * @since 2.0
+		 */
+		Class<?> getContainerClass();
+
+		/**
+		 * @return the index of the type parameter affected by the violated constraint in the container class
+		 *
+		 * @since 2.0
+		 */
+		Integer getContainerElementIndex();
```

Comment:
> The index of the type parameter affected by the violated constraint if stored in a generic container type such as `Optional` or `List`, @code null otherwise.

As discussed on HipChat, I'd rename it to `getTypeArgumentIndex()` as it's only used for generic types, not arrays.